ZConfig/tests/support.py

```diff
@@ -21,21 +21,12 @@
 from ZConfig.loader import ConfigLoader
 from ZConfig.url import urljoin
 
-try:
-    import StringIO
-except ImportError:
-    # Python 3 support.
-    import io as StringIO
-
-try:
-    from urllib import pathname2url
-except ImportError:
-    # Python 3 support.
-    from urllib.request import pathname2url
+from ZConfig._compat import NStringIO as StringIO
+from ZConfig._compat import pathname2url
 
 try:
     __file__
-except NameError:
+except NameError: # pragma: no cover
     import sys
     __file__ = sys.argv[0]
```

Comment:
I'm not sure. I know there is weirdness with `__file__` when the various forms of zipped packages are used, but I don't know exactly what that is for all of them (and there are a bunch now). 

If we can assume the tests are only ever run from an sdist/source checkout, then we could drop this throughout. I just wasn't sure.